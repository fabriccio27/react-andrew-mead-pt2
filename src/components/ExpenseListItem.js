import React from "react";
import {Link} from "react-router-dom";


//aca en vez de usar props, podria usar un destructurado {dispatch(si lo requeria), id, description, amount, createdAt}
const ExpListItem = (props) => {
    return (
        <div>
            <h4><Link to={`/edit/${props.id}`}>{props.description}</Link></h4>
            <p>Amount : {props.amount} | Created:{props.createdAt}</p>
        </div>
    );
};


export default ExpListItem;