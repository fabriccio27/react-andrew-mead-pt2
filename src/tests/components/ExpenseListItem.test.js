import React from "react";
import {shallow} from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import testExpenses from "../fixtures/expenses";

test("should render expense item with data", ()=>{
    const exp = testExpenses[1];
    const wrapper = shallow(<ExpenseListItem key={exp.id} {...exp} />);
    expect(wrapper).toMatchSnapshot();
});
