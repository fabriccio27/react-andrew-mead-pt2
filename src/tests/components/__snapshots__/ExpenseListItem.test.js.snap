// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item with data 1`] = `
<Link
  className="list-item"
  to="/edit/2"
>
  <div>
    <h4
      className="list-item__title"
    >
      Rent
    </h4>
    <span
      className="list-item__subtitle"
    >
      December 27th, 1969
    </span>
  </div>
  <h4
    className="list-item__data"
  >
    $1,095.00
     
  </h4>
</Link>
`;
